src/net_permissions.cpp

```diff
@@ -18,12 +18,28 @@ const std::vector<std::string> NET_PERMISSIONS_DOC{
     "addr (responses to GETADDR avoid hitting the cache and contain random records with the most up-to-date info)"
 };
 
+void NetPermissions::Initialize(NetPermissionFlags& flags, bool whitelist_forcerelay, bool whitelist_relay)
```

Comment:
Passing options individually like this is ugly - worse than just passing `const ArgsManager&` IMO. If you really want to cache it as a bool, maybe make a struct for connection permission options or something to pass instead.

Another possibility is to just cache a single `NetPermissionFlags` for these.

Reply:
I think we can avoid passing them in `Initialize` by setting them in `AddWhitelistPermissionFlags`.

Reply:
I agree to move it to the `CConman::AddWhitelistPermissionFlags` (since the obj has both `whitelist_forcerelay` & `whitelist_relay` vars already set) and you could have something like `CConman:UpdateNetPermissionsFlags`with the logic below.

Reply:
I will address it in the next push.